srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/ValueRetriever.java

```diff
@@ -252,9 +255,85 @@ public void execute(final Patient patient, final VariableEntry variableEntry, fi
         {
             addLabValue(VistaLabs.GLUCOSE, patient, variableEntry, key);
         }
+    },
+    DNR_NOTES
+    {
+        @Override
+        public String retrieveReferenceInfo(final Patient patient, final Variable var)
+        {
+            final StringBuilder variableRefInfo = new StringBuilder();
+            for(final ReferenceNote note: patient.getDnrNotes())
+            {
+                // Add a line break here so that the note body is separated.
+                variableRefInfo.append(String.format("Local Title: %s Sign Date: %s%n%s%n%n",
+                        note.getLocalTitle(), XmlDateAdapter.REFERENCE_NOTE_DATE_FORMAT.print(note.getSignDate()), note.getNoteBody()));
+            }
+            return variableRefInfo.toString();
+        }
     };
     
     /**
+     * An ImmutableSet including any ValueRetrievers that are used to 
+     * add reference information.
+     */
+    public static final ImmutableSet<ValueRetriever> REFERENCE_NOTES_SET = 
+            ImmutableSet.of(DNR_NOTES, ADL_NOTES);
+    /**
+     * An ImmutableSet including any ValueRetrievers that are used for Boolean
```

Comment:
If you're going to make these SortedSets, you need to specify the sort order. (It might be simpler to make them unsorted here and just sort in the view package.)